src/client.rs

```diff
@@ -14,25 +166,101 @@ use std::fmt;
 use std::marker::PhantomData;
 use std::time::Duration;
 
-/// In progress H2 connection binding
+/// In progress HTTP/2.0 connection handshake future.
+///
+/// This type implements `Future`, yielding a `(SendRequest, Connection)`
+/// instance once the handshake has completed.
+///
+/// The handshake is completed once both the connection preface and the initial
+/// settings frame is sent by the client.
+///
+/// The handshake future does not wait for the initial settings frame from the
+/// server.
+///
+/// See [module] level documentation for more details.
+///
+/// [module]: index.html
 #[must_use = "futures do nothing unless polled"]
 pub struct Handshake<T, B: IntoBuf = Bytes> {
     builder: Builder,
     inner: WriteAll<T, &'static [u8]>,
     _marker: PhantomData<B>,
 }
 
-/// Marker type indicating a client peer
+/// Initialize new HTTP/2.0 streams on a connection by sending a request.
+///
+/// This type does no work itself. Instead, it is a handle to the inner
+/// connection state held by [`Connection`]. If the associated connection
+/// instance is dropped, all `SendRequest` functions will error.
+///
+/// See [module] level documentation for more details.
+///
+/// [module]: index.html
+/// [`Connection`]: struct.Connection.html
 pub struct SendRequest<B: IntoBuf> {
     inner: proto::Streams<B::Buf, Peer>,
     pending: Option<proto::StreamKey>,
 }
 
-/// A future to drive the H2 protocol on a connection.
+/// Manages all state associated with an HTTP/2.0 client connection.
```

Comment:
Each short-description uses a slightly different conjugation/structure -- i think it would be clearer to unify these

```
/// In progress HTTP/2.0 connection handshake future.
```

```
/// Initialize new HTTP/2.0 streams on a connection by sending a request.
```

```
/// Manages all state associated with an HTTP/2.0 client connection.
```

I suspect the first two should be changed to the form:

```
/// Holds the state of a pending HTTP/2.0 connection handshake.
```

```
/// Initializes new HTTP/2.0 streams on a connection by sending a request. 
```

... based on what I've seen in other rustdocs...